apps/web/src/utils/importer.ts

```diff
@@ -17,100 +17,62 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { db } from "../common/db";
 import {
-  Note,
-  Notebook,
   ContentType,
-  LegacyNotebook
+  LegacyNotebook,
+  Note,
+  Notebook
 } from "@notesnook-importer/core/dist/src/models";
-import {
-  ATTACHMENTS_DIRECTORY_NAME,
-  NOTE_DATA_FILENAME
-} from "@notesnook-importer/core/dist/src/utils/note-stream";
-import { path } from "@notesnook-importer/core/dist/src/utils/path";
-import { type ZipEntry } from "./streams/unzip-stream";
-import { hashBuffer, writeEncryptedFile } from "../interfaces/fs";
 import { Notebook as NotebookType } from "@notesnook/core";
+import { db } from "../common/db";
+import { writeEncryptedFile } from "../interfaces/fs";
 
-export async function* importFiles(zipFiles: File[]) {
-  const { createUnzipIterator } = await import("./streams/unzip-stream");
-
-  for (const zip of zipFiles) {
-    let count = 0;
-    let filesRead = 0;
-
-    const attachments: Record<string, any> = {};
-
-    for await (const entry of createUnzipIterator(zip)) {
-      ++filesRead;
+export async function pushImportedNoteIntoDatabase(note: Note) {
+  const attachments: Record<string, EncryptedAttachmentFields> = {};
+  await processNoteAttachments(note.attachments, attachments);
```

Comment:
Hm. This looks odd. Why are attachments being passed as a parameter instead of getting returned by the `processNoteAttachments` function?

Reply:
`attachments` were being passed in like this presumably because the process attachment and notes function were in a try catch block. I believe we don't need to handle errors here now as the importer package will do that for us, so II've made the change to return instead